### PR TITLE
feat(remote-host): startChat handler — start a chat from the mobile remote

### PR DESCRIPTION
**Pull request by snakajima**
## What

Adds a `startChat` command-channel handler so the **mobile remote** can start a new chat on the host — either about a whole collection or about one of its records. This brings the desktop's two chat-seeding entry points to the remote:

- collection-level `__MC_VIEW.startChat` (`CollectionView` / custom views)
- the per-record chat box in `CollectionRecordPanel.vue` (`/<slug> id=<itemId> <message>`)

## How

The remote passes `{ message, slug, itemId? }`. The handler:

1. Composes the collection slash command as a prefix:
   - `/{slug} {message}` — whole collection
   - `/{slug} id={itemId} {message}` — one record
2. Spawns a **visible** host chat session (origin `skill`, openable from history) via the existing `spawnSystemWorker`, **fire-and-forget** — no streaming back; starting the chat on the host is enough.
3. Returns `{ started, chatId }`.

Role is intentionally **not** part of the remote API — the caller sends only `message` + `slug` + optional `itemId`; the host runs the chat in its default role (`spawnSystemWorker` → `startChat` requires a concrete `roleId`, so empty is rejected).

Registered in `server/remoteHost/handlers/index.ts` alongside the phase-2 read-only handlers.

### Trust model note

This is the **first write-shaped** remote-host handler (phase-2 handlers were read-only). It stays within the existing uid-scoped Firestore-rules boundary — it only acts on the owner's own workspace — but per the plan's phase-3 caveat, subsequent write/delete/cross-collection handlers should re-evaluate handler-side authz.

## Tests

`test/remoteHost/test_startChat.ts` — 7 tests: prefix composition (with/without `itemId`), param coercion, visible-spawn wiring, missing-slug / blank-message rejection, and spawn-failure surfacing. Spawner stubbed.

## Follow-up (separate repo)

The mobile-remote UI that calls `callHost(channel, "startChat", …)` lives in `mulmoserver` and is not part of this PR.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a new remote-host command to start a visible host chat, with support for collection-wide chats or record-scoped chats.
  * Requests are now routed to the new command handler.
* **Bug Fixes**
  * Improved input validation and normalization for chat start parameters (including message trimming and strict token formatting).
  * Failures during chat creation now surface clearly.
* **Tests**
  * Added unit tests covering chat message formatting, successful chat creation, and error cases for invalid inputs and spawn failures.

### CODE REVIEW

File: server/remoteHost/handlers/startChat.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+export interface StartChatDeps {
+  spawn: typeof spawnSystemWorker;
+}
+
+// Prefix the message with the collection's slash command. `itemId` scopes the
+// chat to one record; empty ⇒ the whole collection. Matches the desktop
+// item-chat format documented in CollectionRecordPanel.vue.
+export const composeMessage = (slug: string, itemId: string, message: string): string => {
+  const prefix = itemId ? `/${slug} id=${itemId}` : `/${slug}`;
```
[the remark this answers is not in the document]
Comment (snakajima) [reply server/remoteHost/handlers/startChat.ts:39]: Good catch — confirmed against `CollectionView.buildChatSeed` (lines 1275–1280): an actual feed (`source === "feed"`) has no `/<slug>` skill command and takes the natural-language `feedChatSeed` seed instead, so `/<slug>` would resolve to nothing.

This PR intentionally scopes to the two desktop **collection** entry points (collection-level + per-record chat), which is what was requested; the mobile remote's feed-chat surface doesn't exist yet (`mulmoserver`). Rather than emit a silently-broken `/<slug>` for feeds, feed support is a follow-up that should port the `feedChatSeed` branch (load the target, and when `source === "feed"` compose the data-path natural-language seed instead of a slash command). Tracking it as deferred; not wiring feed→startChat until then. 👍

File: server/remoteHost/handlers/startChat.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  return `${prefix} ${message}`;
+};
+
+export const createStartChat =
+  (deps: StartChatDeps): CommandHandler =>
+  async (params: JsonObject) => {
+    // Params arrive as JSON over the channel — coerce defensively.
+    const slug = String(params.slug ?? "");
```
[the remark this answers is not in the document]
Comment (snakajima) [reply server/remoteHost/handlers/startChat.ts]: Fixed in 44163f61. `slug`/`itemId` are now validated as trimmed, whitespace-free strings (non-string JSON, whitespace-only, and whitespace-containing values are rejected before composing the prefix), so a malformed seed like `/   hello` can no longer be produced. Added boundary tests for each case.

File: server/remoteHost/handlers/startChat.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    const itemId = params.itemId == null ? "" : String(params.itemId);
+    const message = String(params.message ?? "").trim();
+    if (!slug) throw new Error("slug is required");
+    if (!message) throw new Error("message is required");
```
[the remark this answers is not in the document]
Comment (snakajima) [reply server/remoteHost/handlers/startChat.ts]: Fixed in 44163f61. `slug`/`itemId` are now validated as trimmed, whitespace-free strings (non-string JSON, whitespace-only, and whitespace-containing values are rejected before composing the prefix), so a malformed seed like `/   hello` can no longer be produced. Added boundary tests for each case.

File: test/remoteHost/test_startChat.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  it("omits id= when itemId is absent or null", async () => {
+    const { spawn, calls } = captureSpawn({ ok: true, chatId: "chat-2" });
+    await createStartChat({ spawn })({ slug: "clients", message: "hi" });
+    assert.equal(calls[0].message, "/clients hi");
+  });
+
+  it("throws when slug is missing", async () => {
+    const { spawn } = captureSpawn({ ok: true, chatId: "chat-3" });
+    await assert.rejects(async () => createStartChat({ spawn })({ message: "hi" }), /slug is required/);
+  });
```
[the remark this answers is not in the document]
Comment (snakajima) [reply test/remoteHost/test_startChat.ts:87]: Also add a boundary test — done: whitespace-only slug, whitespace-containing slug, non-string slug, and whitespace-containing itemId are all covered in the updated `test_startChat.ts` (10 tests, all green).

**Comment by snakajima** [comment]
Thanks for the reviews. Pushed 44163f61:

- **slug/itemId validation** (Codex + CodeRabbit): now rejected unless a trimmed, whitespace-free string — no more malformed `/   hello` prefixes. Added boundary tests (whitespace-only / whitespace-containing / non-string slug, whitespace itemId).
- **Feed slugs without a skill command** (Codex P2): confirmed valid; deferred as a follow-up since this PR scopes to the collection/record chat entry points and the mobile feed-chat surface isn't wired yet. Details in the inline reply.

@codex review

**Comment by snakajima** [comment]
Addressed the Codex verdict in 1ae41f49: `test/remoteHost/test_handlers.ts` now includes `startChat` in the registered-handler assertion (previously enumerated only the phase-1/2 handlers).

@codex review

**Comment by snakajima** [comment]
Re: the feed chat-seed point — resolved by decision to **not offer chat for feeds** at all (rather than porting `feedChatSeed`).

94269173: `startChat` now loads the target and refuses when `source === "feed"` (`chat is not available for feeds`), so a stray feed slug can't produce a `/<slug>` seed that resolves to nothing. This is the host-side backstop; the mobile remote won't surface a chat affordance for feeds. Added a unit test asserting a feed is refused without spawning.

@codex review

**Comment by snakajima** [comment]
Addressed the Codex verdict in b1c3fdd4: now that `startChat` loads the target (for feed detection), a `null` result (unknown/stale/deleted slug) is rejected with `collection '<slug>' not found` before spawning — no more `{ started: true }` for a slug that targets nothing. Added an unknown-slug unit test (asserts no spawn). Feeds and unknown slugs are now both refused pre-spawn.

@codex review
